### PR TITLE
Clang format lint action

### PR DESCRIPTION
**Pull request by marinagmoreira**
- Added github action that checks if the repo adheres to the lint check (fixed the script that checks lint errors in the entire repo)
- Added to the pre-commit an automatic clang-format run on the lines that yielded an error - this should correct straightforward errors like spacing without changing the entire file - be aware that clang-format can't correct all the errors automatically"
- Created a clang-format style definition based on google style - this might change in the future as we go.

### CODE REVIEW

**Comment by marinagmoreira** [comment]
Github action passing:
https://github.com/marinagmoreira/astrobee/runs/1113973694

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Check for lint error and auto correct them
+
+name: Check for lint errors
+
+on: [push]
```
Comment (bcoltin) [inline .github/workflows/lint.yaml:5]: not on pull request too?

File: scripts/git/pre-commit [hunk 2]
```diff
@@ -80,12 +82,27 @@ def main():
...
-        for error in output:
-            print "%s:%s: %s" % (filename, str(error[0]), error[1])
+        if len(output) > 0:
+            lines = []
+            for error in output:
+                print "%s:%s: %s" % (filename, str(error[0]), error[1])
+                lines.append(error[0])
+            # Run the clang-format if there are errors in the identified ranges
+            if len(output) > 0:
+                command = "clang-format-8 -style=file -i " + filename
```
Comment (bcoltin) [inline scripts/git/pre-commit:92]: How is clang-format-8 installed? Need to add to install_desktop_packages?

**Review by bcoltin — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (marinagmoreira) [reply .github/workflows/lint.yaml]: Yes that is correct, even though the user will push it to github before doing a pull request, it is useful for the reviewer to see the check. Thanks! I didn't think of that.

File: scripts/git/pre-commit (hunk 2 above)
Comment (marinagmoreira) [reply scripts/git/pre-commit:92]: yes, I added clang-format-8 to the packages_desktop.lst because this is a package that does not go in the robot.

**Review by bcoltin — APPROVED** [review]

**Comment by bcoltin** [comment]
Will wait for @rsoussan to review and then merge.

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,151 @@
+---
+Language:        Cpp
+# BasedOnStyle:  Google
```
Comment (rsoussan) [inline .clang-format:3]: Are all of these options different from Google?  If not, does it work to just override the ones that have changed?

Comment (marinagmoreira) [reply .clang-format:3]: no, just a list of all the options from:
$clang-format-8 -style=google -dump-config > .clang-format
this just dumps all the configurations available, already tuned with the google style.
It's useful to have all the configs to know exactly what are the options available if we need to change anything.
I can add in a comment on which ones were changed from the -dump-config default.

Comment (rsoussan) [reply .clang-format:3]: Ok that would be helpful

Comment (marinagmoreira) [reply .clang-format:3]: I did that in the last commit " highlight the configs that were changed from the original -dump-config in the google style"
thanks!

File: .clang-format [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Attach
+BreakBeforeInheritanceComma: false
+BreakInheritanceList: BeforeColon
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeColon
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     100                           # changed
```
Comment (rsoussan) [inline .clang-format:48]: I think the current linter is 120?

File: .clang-format [hunk 5]
```diff
@@ -0,0 +1,151 @@
+---
+Language:        Cpp
+# BasedOnStyle:  Google
+AccessModifierOffset: -1
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: true              # changed
```
Comment (rsoussan) [inline .clang-format:6]: I think we disagree on this one.  I'd vote to add // clang-format off // clang-format on blocks to areas of new code where custom formatting is desirable, I'm of the opinion that the areas where we would want this are fewer than when we wouldn't, but we should probably all get a vote.  Alternatively, developers could set this as they wish, but then things could switch back and forth in the future based on who has what configs.

File: .clang-format (hunk 4 above)
Comment (marinagmoreira) [reply .clang-format:48]: Oh that's right!

**Review by rsoussan — APPROVED** [review]

**Pull request merged**